ledger/src/blockstore.rs

```diff
@@ -2213,6 +2291,30 @@ impl Blockstore {
         })
     }
 
+    fn get_any_valid_slot_entries(&self, slot: Slot, start_index: u64) -> Vec<Entry> {
+        let (completed_ranges, slot_meta) = self
+            .get_completed_ranges(slot, start_index)
+            .unwrap_or_default();
+        if completed_ranges.is_empty() {
+            return vec![];
+        }
+        let slot_meta = slot_meta.unwrap();
+
+        let entries: Vec<Vec<Entry>> = PAR_THREAD_POOL_ALL_CPUS.with(|thread_pool| {
+            thread_pool.borrow().install(|| {
+                completed_ranges
```

Comment:
Hmm out of curiosity, how big are these ranges? I'm wondering if we'll get better parallel perf if we split up the work of chunks of slots among a thread pool here: https://github.com/solana-labs/solana/pull/10358/files#diff-9ed453cbc107691c26b7e060d05e2ab0R1875, instead of using the thread pool for individual ranges here

Reply:
On MNB, completed_ranges.len() varies from ~80-150. Mostly 100+
Could we try splitting my chunks of slots in a later optimization?

Reply:
ah yeah if the ranges are that large, you probably won't be getting much parallel work as that all goes to one thread

As to the optimization, yeah for sure, I was just curious